### DISCUSSION TITLE
How do xrefs to glossentry topics get type="glossentry"?

### DISCUSSION THREAD

**Discussion by reifsnid** [dita-ot/dita-ot#4461] [Q&A] [answered]
Hi, I don't have an issue, but I am curious how xrefs that reference a glossentry topic, for example:

Topic A
`<xref href="GlossentryTopic.dita">link</xref> `

GlossentryTopic.dita
```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE glossentry PUBLIC "-//OASIS//DTD DITA Glossary Entry//EN" "glossaryentry.dtd">
<glossentry id="GlossentryTopic">
...
</glossentry>
```

end up with `type="glossentry"`, for example:

`<xref href="GlossentryTopicWithGlossentryLink.dita" class="- topic/xref " type="glossentry"><?ditaot usertext?>link</xref>`

in the temp DITA file?

I haven't been able to trace this back to the XSLT that does this.

I ask because I'm using a custom plugin that is acts upon xrefs with `type="glossentry"` and the source DITA file doesn't specify a type for the xref.

Thank you!

Daniel

**Comment by raducoravu** [comment, marked as answer]
The DITA OT has some Java-implemented pre-processing stage steps. Some of these steps call XSLT code, I think that maybe that type attribute computation based on the target might be done in "DITA-OT/plugins/org.dita.base/xsl/preprocess/topicpullImpl.xsl"

**Reply by reifsnid** [reply]
Thanks for your reply @raducoravu! That makes perfect sense. I'm also ashamed to admit that I forgot to check the preprocessing module XSLT. 😅
